gateway.go

```diff
@@ -163,10 +163,39 @@ func (gw *Gateway) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Ms
 	m := new(dns.Msg)
 	m.SetReply(state.Req)
 
+	var ipv4Addrs []net.IP
+	var ipv6Addrs []net.IP
+
+	for _, addr := range addrs {
+		if ipv4 := addr.To4(); ipv4 != nil {
+			ipv4Addrs = append(ipv4Addrs, ipv4)
+		} else {
+			ipv6Addrs = append(ipv6Addrs, addr)
+		}
+	}
+
```

Comment:
I had the same idea, but discarded the idea due to changing it everywhere. Bit if you're fine with changing it everywhere I will have a look into it.

Reply:
I've switched the project to `netip.Addr`.

Reply:
I just thought maybe it'd give us something new but I don't understand it well enough to decide. If you don't think it can make things better or more reliable, then no need to bother.